src/librustc/traits/specialize/specialization_graph.rs

```diff
@@ -291,8 +335,52 @@ impl<'a, 'gcx, 'tcx> Graph {
 
     /// The parent of a given impl, which is the def id of the trait when the
     /// impl is a "specialization root".
-    pub fn parent(&self, child: DefId) -> DefId {
-        *self.parent.get(&child).unwrap()
+    pub fn parent(&self, child: DefId) -> Vec<DefId> {
```

Comment:
Nit: can we rename this to `parents`?

Reply:
Also, the comment should be altered. Something like:

---

Returns the def-id of the parent impl(s) for a given impl -- these are the impls which this impl *specializes*, meaning that it matches a subset of the types that they match.

Reply:
Actually, better to phrase more precisely, I think:

---

Returns the def-id of the parent impl(s) for a given impl. An impl A has a parent impl B if A matches a *strict subset* of the types that B matches.